### ISSUE TITLE
[RFC] Deprecate gradient clipping flags on the trainer

### ISSUE THREAD

**Issue by daniellepintz** [PyTorchLightning/pytorch-lightning#10528]
## Proposed refactoring or deprecation

Deprecate these flags:
https://github.com/PyTorchLightning/pytorch-lightning/blob/7a9a08c5d3ca4699fb439f691c40e1320b37507a/pytorch_lightning/trainer/trainer.py#L122-L123 

since users can already configure gradient clipping on the LM:
https://github.com/PyTorchLightning/pytorch-lightning/blob/7a9a08c5d3ca4699fb439f691c40e1320b37507a/pytorch_lightning/core/lightning.py#L1482-L1515

### Motivation

* There shouldn't be two ways for users to do something
* Reduces # of args on trainer constructor
* Fixes situations like this 
https://github.com/PyTorchLightning/pytorch-lightning/blob/c33df2639f19d49c5e7520294e3221efe402d684/pytorch_lightning/plugins/training_type/deepspeed.py#L432-L433
where we have to reach through LM and trainer to get `gradient_clip_algorithm` property

### Pitch

* Deprecate `gradient_clip_val` and `gradient_clip_algorithm` from trainer constructor and as trainer properties
* Make `gradient_clip_val` and `gradient_clip_algorithm` properties of LM instead

cc @justusschock @awaelchli @akihironitta

**Comment by tchaton** [comment]
Hey @daniellepintz,

Personally, I am not in favour of this depreciation. This would be an important API breaking change and it would make discoverability of this feature harder for new users.
